### ISSUE TITLE
Issue with loading saved scenes

### ISSUE THREAD

**Issue by Peregrine05** [chunky-dev/chunky-denoiser#27]
Documentation about bugs that I have discovered and possible workarounds to those bugs:

# Chunky Denoiser Bugs and Workarounds:
## Bug 1: To replicate it: 
    1. Render the albedo and normal maps, and also the path tracing render for a time.
    2. Pause the render.
    3. Save the scene.
    4. Close Chunky.
    5. Re-open Chunky.
    6. Load the saved scene.
Chunky completely restarts the render, rendering the albedo and normal maps at the default target SPP of 16 for both. Then, Chunky starts rendering the path tracing from the beginning.

## Bug 2: To replicate it:
    1. Render the albedo and normal maps, and also the path tracing render for a time.
    2. Pause the render.
    3. Save the scene.
    4. Close Chunky.
    5. Re-open Chunky.
    6. Load the saved scene.
    7. Reset the render.
    8. Merge render dump from scene directory.
    9. (noted that main target SPP is set to 16 from its previous setting, and the current SPP is the same as what was saved in the render dump.)
    10. Resume the render.
If the current SPP is greater than the target SPP of the albedo map, then chunky starts rendering the normal map. But if the current SPP is less than the target SPP of the normal map, then Chunky starts rendering the normal map from the current SPP saved in the render dump. But, the normal map is rendered on top of the render dump, so the result looks weird; a combination of the normal map and the path tracing render merged into one frame.

![image](https://user-images.githubusercontent.com/5544859/148689435-ca79757f-bd36-4402-b00e-089041b8a029.png)

## Bug 3 (similar to bug 1): To replicate it:
    1. Render without the albedo and normal maps for a time.
    2. Pause the render.
    3. Save the scene.
    4. Close Chunky.
    5. Re-open Chunky.
    6. (noted that the checkboxes for albedo and normal maps are always on by default.)
    7. Load the saved scene.
Chunky not only completely  restarts the render, as in bug 1, but also restarts by rendering the albedo and normal maps before starting the path tracing render from the beginning.

## Bug 4: To replicate it:
    1. Render with or without the albedo and normal maps for a time.
    2. Pause the render.
    3. Save the scene.
    4. Close Chunky.
    5. Re-open Chunky.
    6. Disable albedo and normal maps in an attempt to stop Chunky from restarting the render to render the albedo and normal maps again.
    7. Load the saved scene.
Because the albedo and normal maps are disabled, Chunky does not restart by rendering them. However, it still restarts the path tracing render from the beginning.

## Bug 5: To replicate it:
    1. Render the albedo and normal maps, and also the path tracing render for a time.
    2. Pause the render.
    3. Save the scene.
    4. Close Chunky.
    5. Re-open Chunky.
    6. Disable albedo and normal maps in an attempt to stop Chunky from restarting the render to render the albedo and normal maps again.
    7. Load the saved scene.
    8. After it restarts the path tracing render from the beginning, reset the render.
    9. Merge the render dump for the scene.
    10. Resume the render.
Chunky continues the path tracing render from where it was saved. When it finishes, the denoiser does not use the albedo and normal maps to denoise. The result is a low quality denoised image.

## Workarounds:
For an image without the denoiser:
    1. Render without the albedo and normal maps for a time.
    2. Pause the render.
    3. Save the scene.
    4. Close Chunky.
    5. Re-open Chunky.
    6. Disable the albedo and normal maps.
    7. Load the saved scene.
    8. The path tracing render will start from the beginning. Reset the render.
    9. Merge the render dump from the saved scene.
    10. Resume the render.
Chunky continues the path tracing render from where it was saved.

## For an image with the denoiser:
    1. Render the albedo and normal maps, and also the path tracing render for a time.
    2. Pause the render.
    3. Save the scene.
    4. Close Chunky.
    5. Re-open Chunky.
    6. Disable the albedo and normal maps.
    7. Load the saved scene.
    8. The path tracing render will start from the beginning. Reset the render.
    9. Merge the render dump from the saved scene.
    10. Resume the render.
    11. Enable the albedo and normal maps again.
Chunky continues the path tracing render from where it was saved, and when it finishes, the denoiser uses the albedo and normal maps and denoises properly, making a high quality image.
